frontend/simple/js/events.js

```diff
@@ -3,13 +3,26 @@ import * as Events from '../../../shared/events'
 import store from './state'
 import backend from '../js/backend'
 import {namespace} from '../js/backend/hapi'
+import _ from 'lodash'
 
 export class GroupContract extends Events.HashableGroup {
   static vuex = GroupContract.Vuex({
-    state: { votes: [], payments: [], members: [], invitees: [], profiles: {} },
+    state: { proposals: {}, payments: [], members: [], invitees: [], profiles: {} },
     mutations: {
       HashableGroupPayment (state, {data}) { state.payments.push(data) },
-      HashableGroupVote (state, {data}) { state.votes.push(data) },
+      HashableGroupProposal (state, {data, hash}) { state.proposals[hash] = {...data, for: [], against: []} },
+      HashableGroupVoteForProposal (state, {data}) {
+        if (state.proposals[data.proposalHash]) {
+          state.proposals[data.proposalHash].for.push(data.username)
+          if (state.proposals[data.proposalHash].for.length >= state.proposals[data.proposalHash].threshold) { Vue.delete(state.proposals, data.proposalHash) }
+        }
+      },
+      HashableGroupVoteAgainstProposal (state, {data}) {
+        if (state.proposals[data.proposalHash]) {
+          state.proposals[data.proposalHash].against.push(data.username)
+          if (state.proposals[data.proposalHash].against.length > state.members.length - state.proposals[data.proposalHash].threshold) { Vue.delete(state.proposals, data.proposalHash) }
```

Comment:
- If a line is > 100 please break it up into multiple lines
- Same comment as above re using % threshold